src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs

```diff
@@ -270,16 +270,18 @@ public async Task<IActionResult> Delete(string menuContentItemId, string menuIte
                 return NotFound();
             }
 
+            var menuContentAsJson = (JsonObject)menu.Content;
```

Comment:
`ContentElement.Content` maybe is a type of `JsonDynamicObject` instance

Reply:
Yup, it is
![image](https://github.com/OrchardCMS/OrchardCore/assets/4007293/413eec64-d3dc-400a-8e73-870ae18bcd4d)